src/api/admin.ts

```diff
@@ -0,0 +1,80 @@
+import { DB, ServiceKey } from './types/admin'
+import { setData, getData } from './lib/data'
+import express from 'express'
+import uuid4 from 'uuid/v4'
+
+const app = express.Router()
+
+app.post('/serviceHostsKey', (req, res) => {
+  // create service keys
+  const serviceKeys = getData('serviceKeys')
+  const serviceHostKey: ServiceKey = {
+    id: uuid4(),
+    ...req.body
+  }
+  serviceKeys.push(serviceHostKey)
+  setData('serviceKeys', serviceKeys)
+  res.json(serviceHostKey)
+})
+
+app.get('/serviceHostsKey', (req, res) => {
+  // get all servicekeys
+  const serviceKeys = getData('serviceKeys')
+  res.json(serviceKeys)
+})
+
+app.get('/serviceHostsKey/:id', (req, res) => {
+  // grab one servicekey
+  const serviceKeys = getData('serviceKeys')
+  const selectedKey = serviceKeys.filter(
+    element => element.id === Number(req.params.id)
```

Comment:
`uuid4()` returns a string. If you convert `req.params.id`, you will never find a match.

Reply:
not sure why you are converting all the input params into number, but please correct for all API endpoints.